core/amber/src/main/scala/edu/uci/ics/texera/workflow/common/Marker.scala

```diff
@@ -1,5 +1,50 @@
 package edu.uci.ics.texera.workflow.common
 
+import edu.uci.ics.texera.workflow.common.tuple.Tuple
+import edu.uci.ics.texera.workflow.common.tuple.schema.{AttributeType, Schema, Attribute}
+import scala.collection.mutable
+
 sealed trait Marker
 
+final case class StartOfUpstream() extends Marker
 final case class EndOfUpstream() extends Marker
+
+final case class State(tuple: Option[Tuple] = None, passToAllDownstream: Boolean = false)
```

Comment:
state has a tuple inside??

Reply:
major: this implementation highly depends on Tuple. It can be a temporary solution but I am against this in the short run.

Reply:
The only possible way for the end user to use state now is converting a tuple to state, such that we want to allow converting a tuple to state.